client.cpp

```diff
@@ -76,16 +108,29 @@ int main(int argc, char *argv[]) {
 
     // break up file data into packets and send packets
     int bytes_sent = 0, total = 0;
-    while (total < data_len) {
-        int pkt_size = (int) data_len - total;
-        pkt_size = pkt_size > MAX_PKT_SIZE ? MAX_PKT_SIZE : pkt_size;
-        if ((bytes_sent = sendto(sockfd, data + total, pkt_size, 0, node->ai_addr, node->ai_addrlen)) == -1) {
+    char frame[MAX_FRAME_SIZE];
+    int frame_size;
+    int end = false;
+    // read each section of data from the file, package, and send them 
+    for (int i = 0; i < numBlocks; i++)
+    {
+        
+        int data_size = MAX_DATA_SIZE;
+        if(i == numBlocks - 1){
+            data_size = leftover;
+            end = true;
+        } 
+        // read the data from the file. This should probably later be done in larger chunks
+        src.read(data,data_size);
```

Comment:
You've got a comment here already stating this, but file io is begrudgingly slow and we should read it all in at once into memory.  This line would then be changed to memcpy() from that array into each packet.